.github/workflows/lua-checks.yml

```diff
@@ -0,0 +1,32 @@
+name: Lua Checks
+
+on:
+  workflow_call:
+
+jobs:
+  Lua-Checks:
+    runs-on: ubuntu-24.04
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@1.7.0
+        with:
+          python-version: "3.10"
+          poetry-version: "2.1.2"
+
+      - name: Install Development Environment
+        run: poetry run -- nox -s devenv:install
+
+      - name: Poetry Install
+        run: poetry run -- nox -s devenv:run -- poetry install
+
+      - name: Amalgate Lua Scripts
+        run: poetry run -- nox -s lua:amalgate
+
+      - name: Check if Re-Generated Files Differ From Commit
+        run: git diff --exit-code
+
+      - name: Run Lua Unit Tests
```

Comment:
The amalgate check and the lua unit tests are conceptually independent. Means the amalgate is actually more a check for the python world and the lua unit tests for Lua. Also the Lua tests don't depend on amalgate. I am not sure if it is wise to have both of them in the same workflow.

Reply:
I will split these two.